src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -355,22 +356,35 @@
          (middleware-utils/json-response 200))))
 
 (defn describe-task-file
-  [env-name module-name file]
+  [get-code-content env-name code-id module-name file]
   (let [size (fs/size file)
         sha256 (ks/file->sha256 (io/file file))
-        base-name (fs/base-name file)]
+        base-name (fs/base-name file)
+        ;; we trust the file path from Puppet, so extract the subpath from file
+        static-path (re-find (re-pattern (str #"[^/]+/" module-name "/tasks/" base-name)) file)
+        uri (try
+              ;; if code content can be retrieved, then use static_file_content
+              (when (not= sha256 (ks/stream->sha256 (get-code-content env-name code-id static-path)))
+                (throw (Exception. (i18n/trs "file did not match versioned file contents"))))
+              {:path (str "/puppet/v3/static_file_content/" static-path)
+               :params {:environment env-name :code_id code-id}}
+              (catch Exception e
+                (log/debug (i18n/trs "Static file unavailable for {0}: {1}" file e))
+                {:path (format "/puppet/v3/file_content/tasks/%s/%s" module-name base-name)
+                 :params {:environment env-name}}))]
```

Comment:
just stylistically, do you feel like the uri assignment should be its own function? feels like we're doing a lot in here.

Reply:
Sure, I could go for that.

Reply:
Well, I end up having to pass everything I define in describe-task-file as an argument. So it doesn't feel simpler.

Reply:
Going to leave it unchanged.